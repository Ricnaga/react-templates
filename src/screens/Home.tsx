import { useFormik } from 'formik';
import { useAxios } from '../application/api/axios/useAxios';

enum HomeData {
  USER = 'user',
  PASSWORD = 'password',
}

export function HomeScreen() {
  const {
    functions: { callGETMethod },
  } = useAxios();
  const formik = useFormik({
    initialValues: {
      [HomeData.USER]: '',
      [HomeData.PASSWORD]: '',
    },
    onSubmit: async () => callGETMethod<{ message: string }>({ url: '/user' }),
  });
  return (
    <form onSubmit={formik.handleSubmit}>
      <input
        type="text"
        placeholder="USER"
        name={HomeData.USER}
        onChange={formik.handleChange}
      />
      <input
        type="password"
        placeholder="PASSWORD"
        name={HomeData.PASSWORD}
        onChange={formik.handleChange}
      />
      <button type="submit">ENVIAR</button>
    </form>
  );
}
