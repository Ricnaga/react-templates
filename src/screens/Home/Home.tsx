import { useAxios } from '@application/api/axios/useAxios';
import {
  Tab,
  TabList as ChakraTabList,
  TabPanel,
  TabPanels,
  Tabs,
} from '@chakra-ui/react';
import { useEffect, useState } from 'react';
import { TabCreate } from './components/TabCreate';
import { TabList } from './components/TabList';

enum TabTitle {
  LIST = 'List',
  CREATE = 'Create',
}

export type User = Record<'id' | 'name', string>;

export function HomeScreen() {
  const {
    functions: { callGET },
  } = useAxios();
  const [tabIndex, setTabIndex] = useState(0);
  const [users, setUsers] = useState<Array<User>>([]);

  const getUsers = async () => callGET<Array<User>>({ url: '/user' });

  useEffect(() => {
    getUsers().then((response) => setUsers(response.data));
  }, []);

  const userTabs = [
    { title: TabTitle.LIST, component: <TabList users={users} /> },
    { title: TabTitle.CREATE, component: <TabCreate /> },
  ];

  return (
    <Tabs isFitted onChange={(index) => setTabIndex(index)}>
      <ChakraTabList>
        {userTabs.map(({ title }) => (
          <Tab key={title}>{title}</Tab>
        ))}
      </ChakraTabList>

      <TabPanels>
        {userTabs.map(({ component, title }) => (
          <TabPanel key={title}>{component}</TabPanel>
        ))}
      </TabPanels>
    </Tabs>
  );
}
