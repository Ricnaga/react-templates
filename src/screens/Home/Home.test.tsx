import {
  renderWithTheme,
  screen,
  waitForElementToBeRemoved,
} from '@application/test/testing-library';

import { Home } from './Home';

const callGET = jest.fn().mockImplementation(() =>
  Promise.resolve({
    data: [
      {
        id: '1',
        name: 'mock Name',
      },
    ],
  }),
);

jest.mock('@application/api/axios/useAxios', () => ({
  __esModule: true,
  useAxios: () => ({
    functions: {
      callGET,
    },
  }),
}));

describe('Page: Home', () => {
  it('render correctly', async () => {
    const { container } = renderWithTheme(<Home />);

    await waitForElementToBeRemoved(() => screen.getByText(/carregando/i));

    expect(container).toMatchSnapshot();
  });
});
