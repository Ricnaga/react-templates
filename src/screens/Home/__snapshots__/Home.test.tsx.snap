// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Home render correctly 1`] = `
<div>
  <div
    class="chakra-tabs css-13o7eu2"
  >
    <div
      aria-orientation="horizontal"
      class="chakra-tabs__tablist css-1xhq01z"
      role="tablist"
    >
      <button
        aria-controls="tabs-:r0:--tabpanel-0"
        aria-selected="true"
        class="chakra-tabs__tab css-1pjn7in"
        data-index="0"
        id="tabs-:r0:--tab-0"
        role="tab"
        tabindex="0"
        type="button"
      >
        List
      </button>
      <button
        aria-controls="tabs-:r0:--tabpanel-1"
        aria-selected="false"
        class="chakra-tabs__tab css-1pjn7in"
        data-index="1"
        id="tabs-:r0:--tab-1"
        role="tab"
        tabindex="-1"
        type="button"
      >
        Create
      </button>
    </div>
    <div
      class="chakra-tabs__tab-panels css-8atqhb"
    >
      <div
        aria-labelledby="tabs-:r0:--tab-0"
        class="chakra-tabs__tab-panel css-a5mhaz"
        id="tabs-:r0:--tabpanel-0"
        role="tabpanel"
        tabindex="0"
      >
        <div
          class="css-1269ne9"
        >
          <div
            class="css-0"
          >
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <input
                class="chakra-input css-whnbsr"
                id="field-:r9:"
                placeholder="Insert a name."
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="css-1vlk4q9"
          >
            <div
              class="css-0"
            >
              <div
                class="chakra-card css-15m70g8"
              >
                <div
                  class="chakra-card__body css-1idwstw"
                >
                  <div
                    class="css-k2d005"
                  >
                    <div
                      class="css-tuh9u2"
                    >
                      <p
                        class="chakra-text css-itvw0n"
                      >
                        ID: 
                        1
                      </p>
                    </div>
                    <div
                      class="css-1jr25o7"
                    >
                      <button
                        aria-label="Edit icon"
                        class="chakra-button css-1nn3802"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="chakra-icon css-onkibi"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <g
                            fill="none"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-width="2"
                          >
                            <path
                              d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                            />
                            <path
                              d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
                            />
                          </g>
                        </svg>
                      </button>
                      <button
                        aria-label="Remove icon"
                        class="chakra-button css-1974nvi"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="chakra-icon css-onkibi"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <g
                            fill="currentColor"
                          >
                            <path
                              d="M19.452 7.5H4.547a.5.5 0 00-.5.545l1.287 14.136A2 2 0 007.326 24h9.347a2 2 0 001.992-1.819L19.95 8.045a.5.5 0 00-.129-.382.5.5 0 00-.369-.163zm-9.2 13a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zm5 0a.75.75 0 01-1.5 0v-9a.75.75 0 011.5 0zM22 4h-4.75a.25.25 0 01-.25-.25V2.5A2.5 2.5 0 0014.5 0h-5A2.5 2.5 0 007 2.5v1.25a.25.25 0 01-.25.25H2a1 1 0 000 2h20a1 1 0 000-2zM9 3.75V2.5a.5.5 0 01.5-.5h5a.5.5 0 01.5.5v1.25a.25.25 0 01-.25.25h-5.5A.25.25 0 019 3.75z"
                            />
                          </g>
                        </svg>
                      </button>
                    </div>
                  </div>
                  <p
                    class="chakra-text css-722v25"
                  >
                    Name: 
                    mock Name
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-labelledby="tabs-:r0:--tab-1"
        class="chakra-tabs__tab-panel css-a5mhaz"
        hidden=""
        id="tabs-:r0:--tabpanel-1"
        role="tabpanel"
        tabindex="0"
      >
        <form>
          <div
            class="css-c19cha"
          >
            <div
              class="chakra-form-control css-1kxonj9"
              role="group"
            >
              <label
                class="chakra-form__label css-g6pte"
                for="field-:r6:"
                id="field-:r6:-label"
              >
                Name
              </label>
              <input
                class="chakra-input css-1qrx3zr"
                id="field-:r6:"
                name="name"
                placeholder="Insert a name."
                type="text"
              />
            </div>
            <button
              class="chakra-button css-8bo503"
              type="submit"
            >
              Criar
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
